### ISSUE TITLE
Generator lambda breaks its preceding function

### ISSUE THREAD

**Issue by matejcik** [microsoft/pyright#7779]
**Describe the bug**

The following two declarations typecheck perfectly fine when they are stand-alone.

Placing the lambda after function `foo` will cause `foo` to fail type-checking.

It appears that pyright tries to compare the inferred type of the lambda to the declared type of the preceding function.

```python
def foo() -> None:
    pass

bar = lambda: (yield)
```
output
```
/home/matejcik/projekty/prace/trezor-firmware/core/src/trezor/loop2.py
  /home/matejcik/projekty/prace/trezor-firmware/core/src/trezor/loop2.py:1:18 - error: Return type of generator function must be compatible with "Generator[Any, Any, Any]"
    "Generator[Any, Any, Any]" is incompatible with "None" (reportInvalidTypeForm)
```

**VS Code extension or command-line**
pyright 1.1.360
vscode extension current

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.361, which I just published.
